tests/e2e/editCustodialUserProfileSectionProfile.js

```diff
@@ -0,0 +1,296 @@
+    userProfile.click('@edit');
+    userProfile.click('@diagnosisType');
+    userProfile.click('@diagnosisTypeOther');
+    userProfile.click('@save');
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    userProfile.isSelected('@diagnosisTypeOther', function (result) {
+      this.assert.equal(typeof result, 'object');
+      this.assert.equal(result.status, 0);
+      this.assert.equal(result.value, true);
+    });
+    userProfile.click('@save');
+  },
+  'Edit and save MRN in custodial user profile page': async (browser) => {
+    const userProfilePage = browser.page.userProfilePage();
+    const userProfile = userProfilePage.section.profile;
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    let previousMRN;
+    await userProfile.getValue('@mrn', (result) => {
+      console.log('result', result);
+      previousMRN = result.value;
+    });
+    userProfile.setValue('@mrn', '8192024');
+    userProfile.click('@save');
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    let newMRN;
+    await userProfile.getValue('@mrn', (result) => {
+      console.log('result', result);
+      newMRN = result.value;
+    });
+    userProfile.click('@save');
+    browser.assert.not.strictEqual(previousMRN, newMRN, 'previous mrn and new mrn are not equal');
+    userProfile.click('@edit');
+    userProfile.waitForElementVisible('@mrn', browser.globals.elementTimeout);
+    userProfile.setValue('@mrn', previousMRN);
+    userProfile.click('@save');
+    browser.assert.not.strictEqual(previousMRN, newMRN, 'previous mrn saved in mrn field');
+  },
+  'Edit and save bio in custodial user profile page': async (browser) => {
+    const userProfilePage = browser.page.userProfilePage();
+    const userProfile = userProfilePage.section.profile;
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    let previousBio;
+    await userProfile.getText('@bio', (result) => {
+      console.log('result', result);
+      previousBio = result.value;
+    });
+    userProfile.setValue('@bio', 'test bio');
+    let newBio;
+    userProfile.click('@save');
+    userProfile.waitForElementVisible('@edit', browser.globals.elementTimeout);
+    userProfile.click('@edit');
+    await userProfile.getText('@bio', (result) => {
+      console.log('result', result);
+      newBio = result.value;
+    });
+    browser.assert.not.strictEqual(previousBio, newBio, 'previous bio and new bio are not equal');
+    userProfile.waitForElementVisible('@bio', browser.globals.elementTimeout);
+    userProfile.setValue('@bio', previousBio);
+    userProfile.click('@save');
+    browser.assert.not.strictEqual(previousBio, newBio, 'previous bio saved in bio field');
```

Comment:
**issue (blocking):** This doesn't work as you expected. If you want to again assert if values are changes, you need to reassign them (load it from DOM again). They have still same values from initialization and they are not updated after your actions.

Also - check line 289 - you are asserting, that bios (previous, new) are not same. Great. But then you do same steps (to change bio to previous) and you do same assertion again. Can you explain to me? I think that it is really issue with reassign-ing value.

Reply:
resolved in 7d1886e2f0f2bbb3cbfdfb36453196d314a43ac4 and c5bd6fb2af7883f7cabdc915ebce21be99da11e2

Reply:
I don't think that you resolved it. Check my comment where I described it - https://github.com/tidepool-org/WebUITests/pull/38#discussion_r1732432418